modules/core/test/src/smithy4s/schema/SchemaPartitionSpec.scala

```diff
@@ -0,0 +1,222 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+package schema
+
+import cats.syntax.all._
+import Schema._
+import munit._
+
+final class SchemaPartitionSpec extends FunSuite {
+
+  test("Structure schemas can be divided into partial components") {
+    case class Foo(x: Int, y: Option[Int])
+    val schema = struct(
+      int.required[Foo]("x", _.x),
+      int.optional[Foo]("y", _.y)
+    )(Foo.apply)
+
+    // We're splitting the schema into various components,
+    // which we can use to decode bits of data from various locations
+    // and reconcile later on.
+    val xPartialSchema = schema.partition(_.label == "x")
+
+    // These are two separate documents that the whole data
+    // can't possibly be decoded from.
+    val documentX = Document.obj("x" -> Document.fromInt(1))
+    val documentY = Document.obj("y" -> Document.fromInt(1))
+
+    import SchemaPartition._
+    xPartialSchema match {
+      case SplittingMatch(xSchema, ySchema) =>
+        val decoderX = Document.Decoder.fromSchema(xSchema)
+        val decoderY = Document.Decoder.fromSchema(ySchema)
+
+        val result =
+          (decoderX.decode(documentX), decoderY.decode(documentY)).mapN {
+            PartialData.unsafeReconcile(_, _)
```

Comment:
**suggestion:** it seems like `unsafeReconcile` is order-independent (you can swap the order and it still works), can we have some sort of test for that? Also worth covering in a Scaladoc on the method itself.

Reply:
Addressed in https://github.com/disneystreaming/smithy4s/pull/875/commits/9cb8c6bde034946d6d0332034e385716cf15b9ee